xla/service/hlo_parser_test.cc

```diff
@@ -2541,6 +2541,14 @@ class HloParameterizedParserTest
           original,
           module->ToString(HloPrintOptions().set_print_large_constants(true)));
     }
+    for (HloComputation* computation : module->computations()) {
```

Comment:
I don't think this is the right place to add an assertion. 
Can you add a test below to make sure that the field is null after parsing?